soh/soh/Enhancements/randomizer/settings.cpp

```diff
@@ -532,190 +532,16 @@ void Settings::CreateOptions() {
-        &mTrickOptions[RT_WATER_IRONS_CENTRAL_GS],
-        &mTrickOptions[RT_WATER_CENTRAL_BOW],
-        &mTrickOptions[RT_WATER_HOOKSHOT_FALLING_PLATFORM_GS],
-        &mTrickOptions[RT_WATER_RANG_FALLING_PLATFORM_GS],
-        &mTrickOptions[RT_WATER_RIVER_GS],
-        &mTrickOptions[RT_WATER_DRAGON_JUMP_DIVE],
-        &mTrickOptions[RT_WATER_ADULT_DRAGON],
-        &mTrickOptions[RT_WATER_CHILD_DRAGON],
-        &mTrickOptions[RT_WATER_MQ_CENTRAL_PILLAR],
-        &mTrickOptions[RT_WATER_MQ_LOCKED_GS],
-        &mTrickOptions[RT_LENS_SHADOW],
-        &mTrickOptions[RT_LENS_SHADOW_PLATFORM],
-        &mTrickOptions[RT_LENS_BONGO],
-        &mTrickOptions[RT_SHADOW_UMBRELLA],
-        &mTrickOptions[RT_SHADOW_UMBRELLA_GS],
-        &mTrickOptions[RT_SHADOW_FREESTANDING_KEY],
-        &mTrickOptions[RT_SHADOW_STATUE],
-        &mTrickOptions[RT_SHADOW_BONGO],
-        &mTrickOptions[RT_LENS_SHADOW_MQ],
-        &mTrickOptions[RT_LENS_SHADOW_MQ_INVISIBLE_BLADES],
-        &mTrickOptions[RT_LENS_SHADOW_MQ_PLATFORM],
-        &mTrickOptions[RT_LENS_SHADOW_MQ_DEADHAND],
-        &mTrickOptions[RT_SHADOW_MQ_GAP],
-        &mTrickOptions[RT_SHADOW_MQ_INVISIBLE_BLADES],
-        &mTrickOptions[RT_SHADOW_MQ_HUGE_PIT],
-        &mTrickOptions[RT_SHADOW_MQ_WINDY_WALKWAY],
-        &mTrickOptions[RT_LENS_SPIRIT],
-        &mTrickOptions[RT_SPIRIT_CHILD_CHU],
-        &mTrickOptions[RT_SPIRIT_LOBBY_GS],
-        &mTrickOptions[RT_SPIRIT_LOWER_ADULT_SWITCH],
-        &mTrickOptions[RT_SPIRIT_LOBBY_JUMP],
-        &mTrickOptions[RT_SPIRIT_PLATFORM_HOOKSHOT],
-        &mTrickOptions[RT_SPIRIT_MAP_CHEST],
-        &mTrickOptions[RT_SPIRIT_SUN_CHEST],
-        &mTrickOptions[RT_SPIRIT_WALL],
-        &mTrickOptions[RT_LENS_SPIRIT_MQ],
-        &mTrickOptions[RT_SPIRIT_MQ_SUN_BLOCK_SOT],
-        &mTrickOptions[RT_SPIRIT_MQ_SUN_BLOCK_GS],
-        &mTrickOptions[RT_SPIRIT_MQ_LOWER_ADULT],
-        &mTrickOptions[RT_SPIRIT_MQ_FROZEN_EYE],
-        &mTrickOptions[RT_ICE_BLOCK_GS],
-        &mTrickOptions[RT_ICE_MQ_RED_ICE_GS],
-        &mTrickOptions[RT_ICE_MQ_SCARECROW],
-        &mTrickOptions[RT_LENS_GTG],
-        &mTrickOptions[RT_GTG_WITHOUT_HOOKSHOT],
-        &mTrickOptions[RT_GTG_FAKE_WALL],
-        &mTrickOptions[RT_LENS_GTG_MQ],
-        &mTrickOptions[RT_GTG_MQ_WITH_HOOKSHOT],
-        &mTrickOptions[RT_GTG_MQ_WIHTOUT_HOOKSHOT],
-        &mTrickOptions[RT_LENS_GANON],
-        &mTrickOptions[RT_GANON_SPIRIT_TRIAL_HOOKSHOT],
-        &mTrickOptions[RT_LENS_GANON_MQ],
-        &mTrickOptions[RT_GANON_MQ_FIRE_TRIAL],
-        &mTrickOptions[RT_GANON_MQ_SHADOW_TRIAL],
-        &mTrickOptions[RT_GANON_MQ_LIGHT_TRIAL],
-    });
     mTricksByArea.clear();
     for (int i = 0; i < RT_MAX; i++) {
-        auto& trick = mTrickOptions[i];
-        if (!trick.GetName().empty()) {
-            mTrickNameToEnum[std::string(trick.GetName())] = static_cast<RandomizerTrick>(i);
+        auto trick = &mTrickOptions[i];
```

Comment:
changed declaration because the old way lead to invalid memory references when reference outlived loop into `tricksOption`